sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -391,13 +391,14 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
       case StringType if right.foldable =>
         val sdf = classOf[SimpleDateFormat].getName
         val fString = if (constFormat == null) null else constFormat.toString
-        val formatter = ctx.freshName("formatter")
         if (fString == null) {
           s"""
             boolean ${ev.isNull} = true;
             ${ctx.javaType(dataType)} ${ev.value} = ${ctx.defaultValue(dataType)};
           """
         } else {
+          val formatter = ctx.freshName("sdf")
```

Comment:
Pardon my ignorance but is "sdf" the right name here? I admit I am not familiar with how the code gen works, but just looking at how it seems to operate, formatter seems to be a different reference than sdf.